src/dmd/aliasthis.d

```diff
@@ -64,6 +64,8 @@ extern (C++) final class AliasThis : Dsymbol
 Expression resolveAliasThis(Scope* sc, Expression e, bool gag = false)
 {
     AggregateDeclaration ad = isAggregate(e.type);
+
+Lagain:
```

Comment:
New code should ideally avoid goto. How about we transform this into a `for` loop? Then the only changing thing is `ad`.

Reply:
Also, it seems `ad` is never null inside the loop so you can put it as the test condition and simplify logic inside.